An_attempt_at_Cosine_similarity.py

```diff
@@ -0,0 +1,221 @@
+            t=(key,(dictionary[key]/(len(dictionary))))
+            frequency_list.append(t)
+    return frequency_list
+
+
+#finding the similarities of words between two lists
+"""def comparing(list1,list2):
+    new_list1=[]
+    new_list2=[]
+    for i in list1:
+        for j in list2:
+            if i[1] == j[1]:
+                new_list1.append(i)
+                new_list2.append(j)
+    return new_list1, new_list2
+
+data=comparing(util,commie)
+
+def unique(list1,list2):
+    new_list1=[]
+    new_list2=[]
+    for i[1] in list1:
+        for i[1] not in list2:
+            new_list1.append(i)
+    for j[1] in list2:
+        for j[1] not in list1:
+            new_list2.append(j)
+    return new_list1, new_list2"""
+
+#finds the amount of documents that the word appears in (for 3 documents)
+def IDF(dictionary):
+    amount_of_docs_with_word = []
+    print(dictionary)
+    for key in (dictionary):
+        print(key)
+        a = 0
+        for j in ethics_hist:
+            if key[0]==j[0]:
+                a=+ 1
+        for k in nietzsche_hist:
+            if key[0]==k[0]:
+                a=+ 1
+        for l in utilitarianism_hist:
+            if key[0]==l[0]:
+                a=+ 1
+        t = (dictionary[key],a)
+        amount_of_docs_with_word.append(t)
+    return amount_of_docs_with_word
+
+
+def TF_IDF(dictionary):
+    tf = frequency(dictionary)
+    to_solve_idf = IDF(dictionary)
+    list_of_tfidf = []
+    for key in tf: # for the word that is in both lists
+        if key in to_solve_idf:
+            true_tf = tf[key]
+            true_idf = 1+ math.log(3/to_solve_idf[key]) #solving for the IDF by taking the value of the dict and taking the log of the amount of docs divided by the amount who have the word
+            tf_idf_solved = true_tf*true_idf # multiplying the IDF and TF together
+            t = (dictionary[key],tf_idf_solved)
+            list_of_tfidf.append(t)
+    return list_of_tfidf #returning a list of tuples with TF_IDF and word
+
+def prep_for_cosine(dict1, dict2):
+    list1 = TF_IDF(dict1)
+    list2 = TF_IDF(dict2)
+    list_of_similarities= []
+    list1_tfidt=[]
+    list2_tfidt=[]
+    for i[0] in list1:
+        if i[0] in list2:
+            t=(list1[i[1]], list2[i[1]])
+            list_of_similarities.append(t)
+            list1_tfidt.append((list1[i[1]])**2)
+            list2_tfidt.append((list2[i[1]])**2)
+    return list_of_similarities, list1_tfidt, list2_tfidt # returns a list of the tuples that are the elements of the vectos that will be multiplied together
+
+#finds the cosine similarity of the two docs
+def cosine_similarity(dict1,dict2):
+    list_of_similarities, list1_tfidt, list2_tfidt= prep_for_cosine(dict1, dict2)
+    mag_of_list1 = math.sqrt(sum(list1_tfidt))
+    mag_of_list2= math.sqrt(sum(list2_tfidt))
+    a = 0
+    for i in list_of_similarities:
+        a = a + i[0]*i[1]
+    cosine_sim = a / (mag_of_list1*mag_of_list2)
+    return cosine_sim
+
```

Comment:
I think you could have used better variable names. Some of your variable names, things like `list1_tfidt`, are not quite straightforward. I think your code would be much more readable if those variable names are more straightforward or accompanied by comments describing what they are.